networking_aci/plugins/ml2/drivers/mech_aci/common.py

```diff
@@ -256,6 +256,31 @@ def get_trunk_vlan_usage_on_project(self, context, project_id, segmentation_id=N
 
         return vlan_map
 
+    def get_az_aware_external_subnets(self, context):
+        query = context.session.query(models_v2.Subnet.id, models_v2.Subnet.cidr,
+                                      models_v2.Network.availability_zone_hints, ascope_models.AddressScope.name)
+        query = query.join(models_v2.Network,
+                           models_v2.Network.id == models_v2.Subnet.network_id)
+        query = query.join(extnet_models.ExternalNetwork,
+                           models_v2.Subnet.network_id == extnet_models.ExternalNetwork.network_id)
```

Comment:
I think you could join this on networks instead of subnets which might give a better performance.